### ISSUE TITLE
Add support for various configuration values to pass arbitrary options to pytest.

### ISSUE THREAD

**Issue by nfachan** [maelstrom-software/maelstrom#372]
I was thinking three:
- `--pytest-options` would be passed to both individual test invocations as well as the collect invocation.
- `--pytest-collect-options` would be passed only to the collect invocation.
- `--pytest-test-options` would be passed only to the test invocations. This latter one would be like `--` as specified in #371 , except it would be more explicit and would be a configuration value (thus somebody could set it in a configuration file).

As we don't currently support configuration values that are vectors/lists/arrays, this option would consist of a string that would then be split on spaces. So, to pass two options, `--foo` and `--bar=3`, you'd have to provide a string like `"--foo --bar=3"`.

See #320.

**Definition of Done**
- [ ] `CHANGELOG.md` updated.
- [ ] Documentation in `doc/book/head` updated.
- [ ] Issue is in the correct milestone.

**Comment by bobbobbio** [comment]
I ended up naming these options differently than what is put here in the issue to be consistent with what was done in the other test runners.

In all of them I added the `extra_` prefix to signify that they aren't the complete arguments being passed, but are additional. Also  I used "args" instead of "options" since I thought it made more sense since it accepted any kind of argument, not just things starting with `--`. clap seems to roughly (at least in usage text) refer to everything as arguments but only things starting with `-` as "options".

I also didn't pull the string of having a `--foo=` version of the "var-arg" option, I'll make some separate issue for this that encompasses the other test runners.
